src/app/content/utils.ts

```diff
@@ -9,6 +9,30 @@ export { getBookPageUrlAndParams, getPageIdFromUrlParam, getUrlParamForPageId, t
 export { stripIdVersion } from './utils/idUtils';
 export { scrollSidebarSectionIntoView } from './utils/domUtils';
 
+export const cleanArchiveResponse = (archiveContent: any) => {
+  const chapterSlug = /^chapter-/;
+  const appendixSlug = /^appendix-/;
+  const chapterTitle = 'Chapter';
+  const appendixTitle = 'Appendix';
+
+  if (archiveContent.tree) {
+    archiveContent.tree.contents.map((item: any) => {
+      item.slug = item.slug.replace(chapterSlug, '').replace(appendixSlug, '');
+
+      const domNode = new DOMParser().parseFromString(item.title, 'text/html');
+      const numNode = domNode.querySelector('.os-number');
+
+      if (numNode) {
+        const cleanNumNode = numNode.textContent.replace(chapterTitle, '').replace(appendixTitle, '');
+        item.title = item.title.replace(numNode.textContent, cleanNumNode);
+      }
+
+    });
```

Comment:
all this logic looks good, but you need to recurse by calling `cleanArchiveResponse` on each item